src/providers/github/get-past-sponsors.ts

```diff
@@ -0,0 +1,62 @@
+import { parse } from 'node-html-parser';
+import { $fetch } from 'ohmyfetch'
+
+export type TPublicUser = {
+    username: string
+    avatar: string
+}
+
+export type TPrivateUser = {
+    isPrivate: true
+}
+
+export type TUser = TPublicUser | TPrivateUser
+
+function pickSponsorsInfo(html: string, filter?: (user: TUser) => boolean): TUser[] {
+    const root = parse(html)
+    let sponsors = root.querySelectorAll('div').map(el => {
+        const publicSponsor = el.querySelector('img')
+
+        if (!publicSponsor) {
+            return {
+                isPrivate: true
+            } as TPrivateUser
+        }
+
+        return {
+            username: publicSponsor?.getAttribute('alt')?.replace('@', ''),
+            avatar: publicSponsor?.getAttribute('src')
+        } as TPublicUser
+    })
+
+    if (filter) {
+        sponsors = sponsors.filter(filter)
+    }
+
+    return sponsors
+}
+
+export default async function (username: string, options: {
```

Comment:
Please use the named function with named exports. Also, it would be nice to extract the options type into an interface.